internal/provider/provider.go

```diff
@@ -68,12 +66,12 @@ func (p *ModuleTelemetryProvider) Schema(ctx context.Context, req provider.Schem
 			},
 			"module_source_regex": schema.ListAttribute{
 				ElementType:         types.StringType,
-				Required:            true,
+				Optional:            true,
 				MarkdownDescription: "List of regex as allow list for module source. Only module source that match one of the regex will be collected.",
-				Validators: []validator.List{
-					listvalidators.SizeAtLeast(1),
-					listvalidators.ValueStringsAre(&MustBeValidRegex{}),
-				},
+				//Validators: []validator.List{
```

Comment:
You should be able to keep the validators as they will only apply if the value is no null